app/models/user.rb

```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+    has_many :posts
+    has_many :comments
+    validates :name, presence: true, length: { minimum: 4, maximum: 50 }
```

Comment:
You might want to consider enforcing uniqueness here, as this would allow two Users with the same name to be created:

```
irb(main):006:0> User.where(name: 'johndoe')
  User Load (0.3ms)  SELECT "users".* FROM "users" WHERE "users"."name" = ? LIMIT ?  [["name", "johndoe"], ["LIMIT", 11]]
=> #<ActiveRecord::Relation [#<User id: 1, name: "johndoe", email: "johndoe@example.com", created_at: "2019-09-15 01:19:31", updated_at: "2019-09-15 01:19:31">, #<User id: 2, name: "johndoe", email: "johndoe@example.com", created_at: "2019-09-15 01:19:31", updated_at: "2019-09-15 01:19:31">]>
irb(main):007:0> User.where(name: 'johndoe').count
   (0.4ms)  SELECT COUNT(*) FROM "users" WHERE "users"."name" = ?  [["name", "johndoe"]]
=> 2
```

[OPTIONAL]